app/javascript/components/App.js

```diff
@@ -32,18 +33,28 @@ readApartment = () => {
   render() {
     const { apartments } = this.state
     return(
-    <Router>
+    <BrowserRouter>
       <Header {...this.props} />
       <Routes>
         <Route exact path="/" element={ <Home /> } />
         <Route
             path="/apartmentindex"
             element={<ApartmentIndex apartments={apartments} />}
             />
-        <Route path="/show" element={ <ApartmentShow /> } />
+        <Route path="/apartmentshow/:id" render={(props) =>{
+          let id = props.match.params.id
+          let apartment = this.setState.apartments.find(a => a.id === +id)
+          return <ApartmentShow apartment={apartment} />
+        }}/>
+        {this.props.logged_in &&
+            <Route path="/myapartments" render={(props) => {
+              let apartments = this.state.apartments.filter(a => a.user_id === this.props.current_user.id)
+              return <ProtectedIndex apartments={apartments} />
+            }}/>
+          }
       </Routes>
```

Comment:
I think these will need to be changed back Switch since we are using the older version of react-router.